data/egtea.py

```diff
@@ -15,11 +16,20 @@ class EGATEDataset(Dataset):
     """
 
     def __init__(self, dataset_path, transform=None, target_transform=None):
-        pass
 
+        self.dataset_path = dataset_path
+
+        self.transform = transform
+        self.target_transform = target_transform
+
+        # get dir of all images
+        self.img_list = glob.glob(self.dataset_path)
```

Comment:
One thing i noticed that you assumed was that the user of the function will know the path format suitable with glob and this is high unlikely.
it will be better if you used the `Path` class from `pathlib` to navigate from the root directory ... like
``` (Path(dataset_path) / "Images").glob("*.jpg") ```
where the dataset_path can only be `F:\EGE` (the root of the dataset).